### PR TITLE
feat: flexible notification request handling

### PR DESCRIPTION
**Pull request by xtreem88**
Resolves: https://github.com/customerio/issues/issues/9479

Android apk: https://expo.dev/artifacts/eas/oH3ucFEUnHdY9vpALEA2u8.apk

### CODE REVIEW

File: src/android/withAndroidManifestUpdates.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+export const withAndroidManifestUpdates: ConfigPlugin<
+  CustomerIOPluginOptionsAndroid
+> = (configOuter) => {
+  return withAndroidManifest(configOuter, (props) => {
+    const application = props.modResults.manifest
+      .application as ManifestApplication[];
+    const customerIOMessagingpush =
+      'io.customer.messagingpush.CustomerIOFirebaseMessagingService';
+
+    if (!application[0]['service']) {
```
Comment (mrehan27) [inline src/android/withAndroidManifestUpdates.ts:15]: Question: What does `application[0]` refers to? As per my understanding, it gets the `application` tag from list of manifests where `0` indicates it'll get `application` from app level manifest which is highest in priority. Is this correct?

**Review by mrehan27 — APPROVED** [review]
Just one question, if my assumptions are correct, Android part seems good to me 👍🏻

**Comment by xtreem88** [comment]
> Just one question, if my assumptions are correct, Android part seems good to me 👍🏻

Correct!

File: src/android/withAndroidManifestUpdates.ts (hunk 1 above)
Comment (xtreem88) [reply src/android/withAndroidManifestUpdates.ts:15]: Yes, correct! 
This is a copy of the type definition:
```
export declare type AndroidManifest = {
    manifest: {
        $: {
            'xmlns:android': string;
            'xmlns:tools'?: string;
            package?: string;
            [key: string]: string | undefined;
        };
        permission?: ManifestPermission[];
        'uses-permission'?: ManifestUsesPermission[];
        'uses-permission-sdk-23'?: ManifestUsesPermission[];
        'uses-feature'?: ManifestUsesFeature[];
        application?: ManifestApplication[];
    };
};
```

**Comment by ami-aman** [comment]
I tested the branch on a sample app and found that on `disableNotificationRegistration: true`, the plugin did not add `registerPushNotification` method call in AppDelegate's `didFinishLaunchingWithOptions` but `CIOAppPushNotificationsHandler` class in `PushService.swift` file still had `registerPushNotification` method declaration. 

This was tested using `prebuild` & `prebuild --clean` and both had same results in PushService.swift

**Comment by xtreem88** [comment]
> didFinishLaunchingWithOptions

Fixed!

**Comment by xtreem88** [comment]
> I tested the branch on a sample app and found that on `disableNotificationRegistration: true`, the plugin did not add `registerPushNotification` method call in AppDelegate's `didFinishLaunchingWithOptions` but `CIOAppPushNotificationsHandler` class in `PushService.swift` file still had `registerPushNotification` method declaration.
> 
> This was tested using `prebuild` & `prebuild --clean` and both had same results in PushService.swift

Fixed!

File: README.md [hunk 2]
```diff
@@ -124,6 +124,12 @@ The `customerio-expo-plugin` supports the following configuration options. In mo
             <td><code>undefined</code></td>
             <td>Set the path to your <code>google-services.json</code> file.</td>
         </tr>
+        <tr>
+            <td><code>android.setHighPriorityPushHandler</code></td>
+            <td>boolean</td>
+            <td><code>undefined</code></td>
+            <td>This is optional, if you choose to use a 3rd party plugin to handle notification permissions, but want our SDK to handle the notifications, this sets `io.customer.messagingpush.CustomerIOFirebaseMessagingService` as high priority.</td>
+        </tr>
```
Comment (Shahroz16) [inline README.md:132]: I think we can skip the high priority part because we are putting priority as `-1`.

**Review by Shahroz16 — APPROVED** [review]

**Pull request merged**